### PR TITLE
solution(java): Problems 4, 151, 172, 2186, 287, and 34 - by Raunak

### LINKED ISSUE #69: [PROBLEM] 34. Find First and Last Position of Element in Sorted Array
### Difficulty

Hard

### Problem Description

Given an array of integers `nums` sorted in non-decreasing order, find the starting and ending position of a given `target` value.

If `target` is not found in the array, return `[-1, -1]`.

You must write an algorithm with `O(log n)` runtime complexity.

---

### Example 1:

**Input:** `nums = [5,7,7,8,8,10]`, `target = 8`
**Output:** `[3,4]`

### Example 2:

**Input:** `nums = [5,7,7,8,8,10]`, `target = 6`
**Output:** `[-1,-1]`

### Example 3:

**Input:** `nums = []`, `target = 0`
**Output:** `[-1,-1]`

---

### Constraints:
- $0 <= nums.length <= 105$
- $-109 <= nums[i] <= 109$
- $nums$ is a non-decreasing array.
- $109 <= target <= 109$

### Link

https://leetcode.com/problems/find-first-and-last-position-of-element-in-sorted-array/

### PR DESCRIPTION
**Pull request by oyerounak**
Added more solutions and new approach to some Median and Hard level Questions

### CODE REVIEW

**Review by godkingjay — CHANGES_REQUESTED** [review]
Thank you for your contributions. However there seems to be a problem for your solution in:

### [4. Median of Two Sorted Arrays](https://leetcode.com/problems/median-of-two-sorted-arrays/)

- Java Solution
image

**Comment by oyerounak** [comment]
**_For sure I will make the changes with a more understanding code for beginners._**

File: Hard/4. Median of Two Sorted Arrays/New approach[Arraylist]/Solution.java
Comment (godkingjay) [inline Hard/4. Median of Two Sorted Arrays/New approach[Arraylist]/Solution.java:1]: Don't forget to fix this solution. Thank you.

**Review by oyerounak — COMMENTED** [review]
Here is the solution to Hard/4.Median of Two Sorted Arrays/New approach[add two array]

**Comment by godkingjay** [comment]
> Here is the solution to Hard/4.Median of Two Sorted Arrays/New approach[add two array]

I know. However, it fails some of the test cases once submitted, it passes `2057` out of the `2094` testcases:

image

**Comment by oyerounak** [comment]
`int a = nums1.length;
        int b = nums2.length;
    
        // Calculate the total length of the merged array
        int c = a + b;
    
        // Create a new integer array to hold the merged elements
        int[] arr = new int[c];
    
        // Copy elements from nums1 into arr
        for (int i = 0; i < a; i++) {
            arr[i] = nums1[i];
        }
    
        // Copy elements from nums2 into arr, starting from position a
        for (int i = 0; i < b; i++) {
            arr[a + i] = nums2[i];
        }
    
        // Sort the merged array in ascending order
        Arrays.sort(arr);
    
        // Check if the length of the merged array is even or odd
        if (c % 2 == 0) {
            // Calculate the indices of the two middle elements
            int mid1 = (c - 1) / 2;
            int mid2 = mid1 + 1;
    
            // Calculate the median as the average of the two middle elements
            return (double) (arr[mid1] + arr[mid2]) / 2;
        } else {
            // Calculate the index of the middle element
            int mid = (c - 1) / 2;
    
            // Return the middle element as the median
            return arr[mid];
        }`
        Here is the code for it

**Comment by godkingjay** [comment]
> ```java
>		int a = nums1.length;
> 	int b = nums2.length;
> 
>     // Calculate the total length of the merged array
>     int c = a + b;
> 
>     // Create a new integer array to hold the merged elements
>     int[] arr = new int[c];
> 
>     // Copy elements from nums1 into arr
>     for (int i = 0; i < a; i++) {
>         arr[i] = nums1[i];
>     }
> 
>     // Copy elements from nums2 into arr, starting from position a
>     for (int i = 0; i < b; i++) {
>         arr[a + i] = nums2[i];
>     }
> 
>     // Sort the merged array in ascending order
>     Arrays.sort(arr);
> 
>     // Check if the length of the merged array is even or odd
>     if (c % 2 == 0) {
>         // Calculate the indices of the two middle elements
>         int mid1 = (c - 1) / 2;
>         int mid2 = mid1 + 1;
> 
>         // Calculate the median as the average of the two middle elements
>         return (double) (arr[mid1] + arr[mid2]) / 2;
>     } else {
>         // Calculate the index of the middle element
>         int mid = (c - 1) / 2;
> 
>         // Return the middle element as the median
>         return arr[mid];
>     }
> ```

This solution of yours works, however, you still haven't pushed this solution yet.

The solution that you've pushed is the one that calculates the `mean`.

**Review by godkingjay — APPROVED** [review]
Thank you for your contributions!

**Pull request merged**